### DISCUSSION TITLE
可否导出所有组件的 props 属性对象的定义（非TS类型）

### DISCUSSION THREAD

**Discussion by Thinker-ljn** [tusen-ai/naive-ui#3210] [Q&A] [answered]
我有时需要对 `naive-ui` 的组件二次封装，在使用 `<script setup>` 语法时定义 `props` 属性希望能继承原来的组件的 `props`，让封装的组件也有原来属性的类型提示，但由于 `vue3` 暂时不支持在 `defineProps` 里面使用复杂的 `TS` 类型，所以我想用选项式的 `props` 对象来定义，如下：

```ts
import { avatarProps } from 'naive-ui/lib/avatar/src/Avatar'
const props = defineProps({
  ...avatarProps,
  customProp: {
    type: String,
    default: ''
  }
})
```

![image](https://user-images.githubusercontent.com/8445972/176393908-373c3866-fddc-4184-b325-a78343288a2b.png)

但貌似不是所有的组件都有这个 `props` 属性对象的导出， 如 `Button` 组件就没有导出 `buttonProps`
![image](https://user-images.githubusercontent.com/8445972/176394083-aee64e9b-19a5-440c-9259-0b3d73367753.png)

所以我想询问下，后续是否可以把所有组件的 `props` 属性对象导出？

**Comment by 07akioni** [comment, marked as answer]
我感觉其实是可以的，有兴趣 PR 一个吗

**Reply by Thinker-ljn** [reply]
好的，[https://github.com/TuSimple/naive-ui/pull/3214](https://github.com/TuSimple/naive-ui/pull/3214)
